assets/blocks/quiz/quiz-block/use-update-quiz-has-questions-meta.js

```diff
@@ -16,20 +18,21 @@ import { isQuestionEmpty } from '../data';
  */
 export const useUpdateQuizHasQuestionsMeta = ( clientId ) => {
 	const META_KEY = '_quiz_has_questions';
-	const questionBlocks = useSelect( ( select ) =>
-		select( 'core/block-editor' )
-			.getBlocks( clientId )
-			.filter( ( block ) => ! isQuestionEmpty( block.attributes ) )
-	);
+
+	// It doesn't use the `useSelect` to get the blocks from the main registry.
+	// It avoids getting the blocks from the preview thumbnails.
+	const questionBlocks = dataSelect( blockEditorStore )
```

Comment:
I believe this is the first time I'm seeing a direct call to `select` without putting it inside an `useSelect` call.

From what I know, this basically removes the reactivity behind the `select`, right? I find it interesting that it fixed the issue. :thinking:

Reply:
The main point here is about context. If we use the `useSelect`, it uses the registry from that context specifically (inside the thumbnail). So when we try to get the blocks, it would get the blocks we have in the thumbnail to check. Changing it to the `select` directly, it's getting from the main registry (the real editor content).

Some more context:

- https://github.com/WordPress/gutenberg/blob/trunk/packages/data/src/components/use-select/index.js
- https://developer.wordpress.org/block-editor/reference-guides/packages/packages-data/#useregistry